src/Nox.Generator/Application/Dto/EntityUpsertDtoGenerator.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Nox.Generator.Common;
+using Nox.Solution;
+using Nox.Types;
+using Nox.Types.Extensions;
+
+namespace Nox.Generator.Application.Dto;
+
+internal class EntityUpsertDtoGenerator : INoxCodeGenerator
+{
+    public NoxGeneratorKind GeneratorKind => NoxGeneratorKind.Domain;
+
+    public void Generate(
+      SourceProductionContext context,
+      NoxCodeGenConventions codeGeneratorState,
+      GeneratorConfig config,
+      Action<string> log,
+      string? projectRootPath
+      )
+    {
+        NoxSolution solution = codeGeneratorState.Solution;
+        context.CancellationToken.ThrowIfCancellationRequested();
+
+        if (solution.Domain is null || !solution.Domain.Entities.Any())
+            return;
+
+        foreach (var entity in codeGeneratorState.Solution.Domain!.Entities.Where(entity => entity.IsOwnedEntity))
+        {
+            var attributes = entity.Attributes ?? Enumerable.Empty<NoxSimpleTypeDefinition>();
```

Comment:
```suggestion
            var attributes = entity.Attributes;
```

You don't need to have a null check, because it would instantiate this property with a default value which is an empty array

Reply:
Updated